### ISSUE TITLE
snmpwalk crashes the server

### ISSUE THREAD

**Issue by probert100** [delimitry/snmp-server#17]
Looks like after the last bugfix the snmpwalk stopped working:

running  `snmpwalk -v 2c -c public 0.0.0.0:161 .1.3.6.1.4.1`
crashes the server 

``` ./snmp-server.py -c ./config.py 
SNMP server listening on 0.0.0.0:161
Traceback (most recent call last):
  File "./snmp-server.py", line 920, in <module>
    main()
  File "./snmp-server.py", line 914, in main
    snmp_server(host, port, oids)
  File "./snmp-server.py", line 854, in snmp_server
    error_status, error_index, oid, oid_value = handle_get_next_request(oids, oid)
UnboundLocalError: local variable 'oid' referenced before assignment
```

**Comment by delimitry** [comment]
I've fixed it. Please verify.

**Comment by probert100** [comment]
Great! snmpwalk works! 
Thank you for fixing it so quickly.

One last thing that I was wondering if you considered implementing snmpbulkget
Currently, when used:
`snmpbulkget -v 2c -Cr2 -c public  localhost .1.3.6.1.4.1` 

Server prints in debug error, over and over until client times out.
The snmpbulkget uses request-id of 165

```SNMP server listening on 0.0.0.0:161
[DEBUG] Received 40 bytes from ('127.0.0.1', 65048)
[DEBUG] ASN1_SEQUENCE: length = 38
[DEBUG] ASN1_INTEGER: 1
[DEBUG] ASN1_OCTET_STRING: public
[ERROR] SNMP request parsing failed: Invalid tag for PDU unit "request-id"
[DEBUG] Received 40 bytes from ('127.0.0.1', 65048)
[DEBUG] ASN1_SEQUENCE: length = 38
[DEBUG] ASN1_INTEGER: 1
[DEBUG] ASN1_OCTET_STRING: public
[ERROR] SNMP request parsing failed: Invalid tag for PDU unit "request-id"
[DEBUG] Received 40 bytes from ('127.0.0.1', 65048)
[DEBUG] ASN1_SEQUENCE: length = 38
[DEBUG] ASN1_INTEGER: 1
[DEBUG] ASN1_OCTET_STRING: public
[ERROR] SNMP request parsing failed: Invalid tag for PDU unit "request-id"
[DEBUG] Received 40 bytes from ('127.0.0.1', 65048)
[DEBUG] ASN1_SEQUENCE: length = 38
[DEBUG] ASN1_INTEGER: 1
[DEBUG] ASN1_OCTET_STRING: public
[ERROR] SNMP request parsing failed: Invalid tag for PDU unit "request-id"
[DEBUG] Received 40 bytes from ('127.0.0.1', 65048)
[DEBUG] ASN1_SEQUENCE: length = 38
[DEBUG] ASN1_INTEGER: 1
.
.
.
```

**Comment by delimitry** [comment]
Great! :+1: 
And `snmpbulkget` implemented in #19 by @probert100
